abdulbasit-hussein-yazen/src/Components/SearchBox.js

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from "react";
+import { Form, FormControl, Button, Spinner } from "react-bootstrap";
+function SearchBox(props) {
+    const [searchText, setSearchText] = useState("");
+    const [query, setQuery] = useState("");
+    const [isLoading, setIsLoading] = useState(false);
+    const TMDB_BASE_URL = "https://api.themoviedb.org/3";
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        setQuery(searchText);
+        const constructUrl = (path, query) => {
+            return `${TMDB_BASE_URL}/${path}?api_key=${atob(
+                "ZDJmYTdhZDFlMjZhZjA4NDdkMzQ5ZDdkYmQ1ZjkzZTU="
+            )}&query=${query}`;
+        };
+
+        fetch(constructUrl("search/movie", searchText))
+            .then((response) => response.json())
+            .then((data) => {
+                props.setMovies(data.results);
```

Comment:
Awesome job on lifting the state 2 stages. This is very well done. Hats off to you! 🎩